micrOS/source/LM_OV2640.py

```diff
@@ -130,7 +130,7 @@ def percent_to_value(percent):
     return 'Settings applied.'
 
 
-def capture():
+async def capture():
```

Comment:
Why async? I assume we need syncronous capture function as well for /cam/snaphot endpoint.

Reply:
For sound recording on i2s, asyncio.StreamReader is used to avoid blocking when reading large number of samples. Because of that, the HTTP callback function must be async. I assumed it would not change much to change the callback functions to be async. The other solution I guess would be to check if the callback function is async with inspect.iscoroutinefunction(...), when the callback function is invoked at micrOS/source/Server.py#245

Reply:
I'm okay with either solution.

Reply:
sadly inspect is not available in micropython as I know. It seems me a corner case for web endpoints, because for all the other case it works as sync scope. Of course async can be better in performance. We can also detect from the dtype string the callback should be called as sync or async way. And it is probably one more optional string in dtype: "text/html/async" or "text/html/sync" and default can be sync (if last /param missing). Or overthinking? :D (I would avoid unnecessary asyncio lib imports overall, when possible)

Reply:
Somehow I forgot to check if inspect is available :). You are right, for later use, it would be more beneficial if callback functions are synchronous and having an async callback is a corner case. Probably passing back additional information about the function being sync/async is the best solution.

Reply:
I was wondering maybe a really short optional ending if dtypes are fine like, :a or :s because easy to check if we have : in dtype and we can cut it out use the dtype as today and run the callback as we indicated :a (async) or :s (sync). Just an idea for the implementation to keep everything efficient and generic.

Reply:
seems good to me